mod-source-record-manager-server/src/main/java/org/folio/dao/JobExecutionDaoImpl.java

```diff
@@ -451,4 +460,64 @@ private String buildOrderByClause(List<SortField> sortFields) {
       .collect(Collectors.joining(", ", "ORDER BY ", EMPTY));
   }
 
+  @Override
+  public Future<RowSet<Row>> hardDeleteJobExecutions(String tenantName, long diffNumberOfDays) {
+    Promise<RowSet<Row>> promise = Promise.promise();
+    Promise<SQLConnection> connection = Promise.promise();
+    PostgresClient postgresClient = pgClientFactory.createInstance(tenantName);
+    String rollbackMessage = "Rollback transaction. Error during jobExecution deletion process.";
+    Future.succeededFuture()
+      .compose(c -> {
+        postgresClient.startTx(connection);
+        return connection.future();
+      })
+      .compose(s -> fetchJobExecutionIdsToBeConsideredForDeleting(tenantName, diffNumberOfDays, connection, postgresClient))
+      .compose(rowSet -> {
+        if (rowSet.rowCount() < 1) {
+          throw new NotFoundException(rollbackMessage);
+        }
+        return mapRowsetValuesToListOfString(rowSet);
+      })
+      .compose(strings -> {
```

Comment:
I think these tables need to be deleted in a specific order due to foreign key constraints. If not then maybe delete from the "top" should cascade all the way down to child tables, which would mean just one delete statement instead of many. I definitely see the benefit of deleting child tables first. What do you think?

Reply:
Data from the main table job_execution are deleting in the last turn. During testing we did not observe any foreign key constraints, data from all 5 tables successfully deleted. But if you think we need to change order for some tables - let us know and we will retest it.

Reply:
Looking one more time, theoretically future to delete data from job_execution table can occur earlier, because its Composite Future, and Im not sure that order passing to Composite future is equals to order of future's execution, so potentially it can lead to Foreign Keys constraints, need to think

Reply:
Refactored to delete firstly from 4 related tables and after this from main job_execution table. This approach will prevent foreign key constraint exceptions.